includes/forms/controllers/class.llms.controller.account.php

```diff
@@ -295,8 +295,17 @@ public function reset_password() {
 			return $user;
 		}
 
-		// Reset the password.
-		reset_password( $user, llms_filter_input( INPUT_POST, 'password', FILTER_SANITIZE_STRING ) );
+		/**
+		 * Reset the password.
+		 *
+		 * We're doing this identically to how the WP Core does as of the writing of this comment.
+		 *
+		 * I feel like it should be sanitized but when slashes are used in a password any amount of sanitization
+		 * causes the reset to save but then, later, when trying to login users can't.
+		 *
+		 * @link https://github.com/gocodebox/lifterlms/issues/1567
+		 */
+		reset_password( $user, $_POST['password'] ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotValidated, WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
```

Comment:
I've investigated a little bit into this, just to bear this in mind when we use the PHP `filter_input()`
I've added this debugging code just before this line:
```
		error_log( sprintf( '$_POST[\'password\']: %s', $_POST['password'] ) );
		error_log( sprintf( 'Filter input INPUT_POST, FILTER_SANITIZE_STRING: %s', filter_input( INPUT_POST, 'password', FILTER_SANITIZE_STRING ) ) );
		error_log( sprintf( 'Filter input INPUT_POST, FILTER_UNSAFE_RAW: %s', filter_input( INPUT_POST, 'password', FILTER_UNSAFE_RAW ) ) );
		error_log( sprintf( 'filter_var $_POST[\'password\'], FILTER_SANITIZE_STRING: %s', filter_var( $_POST[ 'password' ], FILTER_SANITIZE_STRING ) ) );
		error_log( sprintf( 'filter_var $_POST[\'password\'], FILTER_UNSAFE_RAW: %s', filter_var( $_POST[ 'password' ], FILTER_UNSAFE_RAW ) ) );
```

and here's the result:

```
[18-Mar-2021 12:12:00 UTC] $_POST['password']: \\a!\"£$%&/()=?\'^*#@-:.;,<>_
[18-Mar-2021 12:12:00 UTC] Filter input INPUT_POST, FILTER_SANITIZE_STRING: \a!&#34;£$%&/()=?&#39;^*#@-:.;,_
[18-Mar-2021 12:12:00 UTC] Filter input INPUT_POST, FILTER_UNSAFE_RAW: \a!"£$%&/()=?'^*#@-:.;,<>_
[18-Mar-2021 12:12:00 UTC] filter_var $_POST['password'], FILTER_SANITIZE_STRING: \\a!\&#34;£$%&/()=?\&#39;^*#@-:.;,_
[18-Mar-2021 12:12:00 UTC] filter_var $_POST['password'], FILTER_UNSAFE_RAW: \\a!\"£$%&/()=?\'^*#@-:.;,<>_

```
Can you see the "problem"? Aside from the fact that the string sanitization encodes some chars (and we should avoid that)...
Basically the $_POST is slashed, see the first and the latter 2 prints.  While `filter_input()` works on the raw data.
That's why even using the default `FILTER_UNSAFE_RAW` doesn't work when using `filter_input()`.
Why the global `$_POST` is slashed though? Because...
https://github.com/WordPress/WordPress/blob/5.7/wp-settings.php#L429